packages/api/src/controllers/access-control.ts

```diff
@@ -88,11 +88,21 @@ async function fireGateWebhook(
     resp = await fetchWithTimeoutAndRedirects(webhook.url, params);
     statusCode = resp.status;
 
-    if (resp.status < 200 || resp.status >= 300) {
+    // Dispose of the response body
+    respBody = await resp.text();
+    if (resp.status >= 200 && resp.status < 300) {
+      // try parsing response body to check for access denied with a 200
+      const respCode = checkRespBody(respBody);
+      if (respCode) {
+        statusCode = respCode;
+      }
+    }
+
+    if (statusCode < 200 || statusCode >= 300) {
       console.error(
-        `access-control: gate: webhook=${webhook.id} got response status != 2XX statusCode=${resp.status}`
+        `access-control: gate: webhook=${webhook.id} got response status != 2XX statusCode=${statusCode}`
```

Comment:
It'll mess up your nice neat code, but it might be best to log the real response code and the body one, since they have different meanings